### PR TITLE
blockheader rules

### PR DESCRIPTION
**Pull request by nandotorterolo**
## Purpose
blockheader rules
## Testing
```scala
package co.topl.consensus.models

object BlockHeaderValidator extends scalapb.validate.Validator[co.topl.consensus.models.BlockHeader] {
  def validate(input: co.topl.consensus.models.BlockHeader): scalapb.validate.Result =
    co.topl.consensus.models.BlockIdValidator.validate(input.parentHeaderId) &&
    co.topl.consensus.models.EligibilityCertificateValidator.validate(input.eligibilityCertificate) &&
    co.topl.consensus.models.OperationalCertificateValidator.validate(input.operationalCertificate) &&
    scalapb.validate.Result.run(io.envoyproxy.pgv.BytesValidation.length("BlockHeader.address", input.address, 32))
  
}
```

## Tickets
* closes https://topl.atlassian.net/browse/BN-829

### CODE REVIEW

File: proto/consensus/models/block_header.proto [hunk 1]
```diff
@@ -24,13 +28,29 @@ message BlockHeader {
...
   uint64 slot = 7;
   // A certificate indicating that the block producer was eligible to make this block
-  EligibilityCertificate eligibilityCertificate = 8;
+  EligibilityCertificate eligibilityCertificate = 8 [(validate.rules).message.required = true];
   // A certificate indicating the operator's commitment to this block
-  OperationalCertificate operationalCertificate = 9;
+  OperationalCertificate operationalCertificate = 9 [(validate.rules).message.required = true];
   // Optional metadata stamped by the operator.  Must be latin-1 encoded, and must be at most 32 bytes in length.
   // optional
   bytes metadata = 10;
```
Comment (SeanCheatham) [inline proto/consensus/models/block_header.proto:36]: metadata could include `[(validate.rules).bytes.max_len = 32]`

File: proto/consensus/models/block_header.proto [hunk 2]
```diff
@@ -6,11 +6,15 @@ import 'consensus/models/block_id.proto';
 import 'consensus/models/eligibility_certificate.proto';
 import 'consensus/models/operational_certificate.proto';
 
+import "validate/validate.proto";
+import "scalapb/scalapb.proto";
+import "scalapb/validate.proto";
+
```
Comment (SeanCheatham) [inline proto/consensus/models/block_header.proto:12]: txRoot and bloomFilter could also be constrained to `32` and `256` bytes, respectively.

**Review by SeanCheatham — APPROVED** [review]

File: proto/consensus/models/block_header.proto (hunk 1 above)
Comment (nandotorterolo) [reply proto/consensus/models/block_header.proto:36]: Thanks for reviewing it

File: proto/consensus/models/block_header.proto (hunk 2 above)
Comment (nandotorterolo) [reply proto/consensus/models/block_header.proto:12]: Thanks for reviewing it

**Pull request merged**